Mathlib/Combinatorics/SimpleGraph/Triangle/Tripartite.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.SimpleGraph.Triangle.Basic
+
+/-!
+# Construct a tripartite graph from its triangles
+
+This file contains the construction of a simple graph on `α ⊕ β ⊕ γ` from a list of triangles
+`(a, b, c)` (with `a` in the first summand, `b` in the second, `c` in the third).
+
+We call
+* `t : Finset (α × β × γ)` the set of *triangle indices* (its elements are not triangles within the
+  graph but instead index them).
+* *explicit* a triangle of the constructed graph coming from a triangle index.
+* *accidental* a triangle of the constructed graph not coming from a triangle.
+
+The two important properties of this construction are:
+* `SimpleGraph.TripartiteFromTriangles.ExplicitDisjoint`: Whether the explicit triangles are
+  edge-disjoint.
+* `SimpleGraph.TripartiteFromTriangles.NoAccidental`: Whether all triangles are explicit.
+
+This construction shows up unrelatingly twice in the theory of Roth numbers:
+* The lower bound of the Ruzsa-Szemerédi problem: From a Salem-Spencer set `s` on a commutative ring
```

Comment:
The terminology 'Salem-Spencer set' is not really used in the literature, and is perhaps confusing - better to say just 'a set without three-term arithmetic progressions'.

Reply:
Opened #12682 to fix the existing mentions of the name.